tests/test_session.py

```diff
@@ -372,6 +372,59 @@ async def test_session_communication_with_a_udp_echo_server(
     assert await src_peer.api.session_close_client(protocol='udp', bound_ip='127.0.0.1', bound_port=src_sock_port) is True
     assert len(await src_peer.api.session_list_clients('udp')) == 0
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize("src,dest", random_distinct_pairs_from(barebone_nodes(), count=PARAMETERIZED_SAMPLE_SIZE))
+async def test_session_communication_with_udp_loopback_service(
+        src: str, dest: str, swarm7: dict[str, Node]
+):
+    """
+    HOPR UDP socket buffers are set to 462 bytes to mimic the underlying MTU of the HOPR protocol.
+    """
+
+    packet_count = 100 if os.getenv("CI", default="false") == "false" else 50
+    expected = [f"{i}".rjust(HOPR_SESSION_MAX_PAYLOAD_SIZE) for i in range(packet_count)]
```

Comment:
Would it be possible to implement the stress test in `test_stress.py` in terms of the self looping as well?

Reply:
Also, IMHO, stress test should use different node as Entry and Exit

Reply:
But that is what happens today anyway. There's just an extra echo server in between.